model/src/main/kotlin/OrtResult.kt

```diff
@@ -245,17 +245,22 @@ data class OrtResult(
 
     /**
      * Return a map of all de-duplicated [Issue]s associated by [Identifier]. If [omitExcluded] is set to true, excluded
-     * issues are omitted from the result.
+     * issues are omitted from the result. If [omitResolved] is set to true, resolved issues are omitted from the
+     * result.
      */
     @JsonIgnore
-    fun getIssues(omitExcluded: Boolean = false): Map<Identifier, Set<Issue>> {
+    fun getIssues(omitExcluded: Boolean = false, omitResolved: Boolean = false): Map<Identifier, Set<Issue>> {
         val analyzerIssues = analyzer?.result?.getAllIssues().orEmpty()
         val scannerIssues = scanner?.getAllIssues().orEmpty()
         val advisorIssues = advisor?.results?.getIssues().orEmpty()
 
         val allIssues = analyzerIssues.zipWithCollections(scannerIssues).zipWithCollections(advisorIssues)
 
-        return allIssues.filterKeys { id -> !omitExcluded || !isExcluded(id) }
+        return allIssues.mapValues { (_, issues) ->
+            issues.filterTo(mutableSetOf()) { !omitResolved || !isResolved(it) }
+        }.filter { (id, issues) ->
+            issues.isNotEmpty() && (!omitExcluded || !isExcluded(id))
```

Comment:
IMO it's not so nice that `issues.isNotEmpty()` is required here to "clean up" after the first filtering, and I'm wondering whether there's a more elegant way to only filter / map once.

Reply:
This is now addressed in the next commit, where I took your sniplet with `mapNotNull`